hummingbot/connector/gateway_EVM_AMM.py

```diff
@@ -603,63 +599,61 @@ async def update_order_status(self, tracked_orders: List[GatewayInFlightOrder]):
-                        event_tag,
-                        event_class(
-                            timestamp=self.current_timestamp,
-                            order_id=tracked_order.client_order_id,
-                            base_asset=tracked_order.base_asset,
-                            quote_asset=tracked_order.quote_asset,
-                            fee_asset=tracked_order.fee_asset,
-                            base_asset_amount=tracked_order.executed_amount_base,
-                            quote_asset_amount=tracked_order.executed_amount_quote,
-                            fee_amount=fee,
-                            order_type=tracked_order.order_type,
-                            exchange_order_id=tracked_order.exchange_order_id
-                        )
+                )
+                tracked_order.last_state = "FILLED"
+                self.logger().info(f"The {tracked_order.trade_type.name} order "
+                                   f"{tracked_order.client_order_id} has completed "
+                                   f"according to order status API.")
+                event_tag: MarketEvent = (
+                    MarketEvent.BuyOrderCompleted if tracked_order.trade_type is TradeType.BUY
+                    else MarketEvent.SellOrderCompleted
+                )
+                event_class: Union[Type[BuyOrderCompletedEvent], Type[SellOrderCompletedEvent]] = (
+                    BuyOrderCompletedEvent if tracked_order.trade_type is TradeType.BUY
+                    else SellOrderCompletedEvent
+                )
+                self.trigger_event(
+                    event_tag,
+                    event_class(
+                        timestamp=self.current_timestamp,
+                        order_id=tracked_order.client_order_id,
+                        base_asset=tracked_order.base_asset,
+                        quote_asset=tracked_order.quote_asset,
+                        base_asset_amount=tracked_order.executed_amount_base,
+                        quote_asset_amount=tracked_order.executed_amount_quote,
+                        order_type=tracked_order.order_type,
+                        exchange_order_id=tracked_order.exchange_order_id
                     )
-                else:
-                    self.logger().info(
-                        f"The market order {tracked_order.client_order_id} has failed according to order status API. ")
-                    self.trigger_event(MarketEvent.OrderFailure,
-                                       MarketOrderFailureEvent(
-                                           self.current_timestamp,
-                                           tracked_order.client_order_id,
-                                           tracked_order.order_type
-                                       ))
+                )
+                self.stop_tracking_order(tracked_order.client_order_id)
+            elif update_result["txStatus"] in [-1, 1]:
```

Comment:
`txStatus` set to -1 simply means the transaction hasn't been mined yet. It should not be treated as a failure.